### ISSUE TITLE
Is this project no longer being actively maintained?

### ISSUE THREAD

**Issue by Zamiell** [martynsmith/node-irc#399]
With 52 open issues, 32 open pull requests, and the last code change being 4+ months ago (other than a commit fixing a typo in the README.md file)...

Is it time to start talking about having some new people take the reigns?

**Comment by jirwin** [comment]
I'm all for that. Honestly this project should probably be deprecated. There are a lot of improvements to be made, but it is becoming a big pile of spaghetti code, and is very difficult to maintain. Almost every patch introduces new bugs for someone else.

**Comment by Zamiell** [comment]
As a collaborator, do you have any recommendations for an alternative?

**Comment by morenoh149** [comment]
@Zamiell you could fork it and merge in all the prs you want. There are no alternative afaict. But I'd look at https://github.com/oftn/oftn-bot and try to extract what you need

**Comment by jirwin** [comment]
Here are two that I've played with.

https://github.com/gf3/IRC-js
https://github.com/twitch-irc/twitch-irc

**Comment by IanMitchell** [comment]
First one hasn't been updated in a year, too. If PRs start getting merged in again, might be worth focusing on code cleanup for this lib. Still seems to be the best option out there

**Comment by morenoh149** [comment]
@IanMitchell but like, no one steps up so ...

**Comment by IanMitchell** [comment]
Yeah, but that might be because PRs are inactive - a lot are outstanding with no comments on them. Why put the effort in if it'll just sit in a branch forever?

**Comment by Jnull** [comment]
With a marked name like 'irc' in node: "npm install irc", i think its worthwhile to revive, this will always generate lots of traffic.  

Is it possible to rollback to a good collaborative revision and re-implement? 
Would it be worth setting up a small ircd for testing and self service user support?

**Comment by morenoh149** [comment]
the `irc` name on npm can be released if you ask. It's at their discretion. But perhaps this project could voluntarily release if it's really not going to be maintained. I maintain another project that I didn't found. The original author lost interest. PRs get merged if the proposer defends a line of inquiry. That's my litmus test. But @jirwin is describing that contributions are adding complexity which would be another issue.

**Comment by Jnull** [comment]
It sounds like a good problem to have with so many contributions. I just hope I'm not to late to the party, this seems to be the most user friendly project of its kind, even out of the ones just previously mentioned. 

I suggest the name be kept as long as possible though, I'm sure a lot of people still depend on it.

**Comment by jirwin** [comment]
I'll start heading through the PRs and merge what I can. A big issue we've had is people submitting PRs, and never coming back to respond to comments. It is hard for me to make the time fix up every PR that comes in. I'll make a goal to have the backlog as clean as I can get it, and publish a new release this coming Monday(9/7).

**Comment by Jnull** [comment]
Jirwin, anything i can do?

**Comment by Zamiell** [comment]
Fantastic jirwin, thanks so much.

**Comment by jirwin** [comment]
@Jnull It'd be easier to coordinate if you joined ##node-irc on freenode so we could chat.

**Comment by Jnull** [comment]
@jirwin Just a check-in.

**Comment by ismell** [comment]
I would close all the PRs that are abandoned.

**Comment by Throne3d** [comment]
It seems to have been maintained a little in the meantime, but it still has a few major issues and a lot of unmerged pull requests (possibly broken?).

I've started a fork at https://github.com/Throne3d/node-irc which has already fixed a few major issues, without (I think) introducing new bugs.
